src/web_interface/components/analysis_routes.py

```diff
@@ -167,12 +167,16 @@ def post_update_analysis(self, uid, re_do=False):
         return render_template('upload/upload_successful.html', uid=uid)
 
     def _schedule_re_analysis_task(self, uid, analysis_task, re_do):
-        fw = convert_analysis_task_to_fw_obj(analysis_task)
         if re_do:
+            olf_fw_obj = None
             with ConnectTo(AdminDbInterface, self._config) as sc:
                 sc.delete_firmware(uid, delete_root_file=False)
+        else:
+            with ConnectTo(FrontEndDbInterface, self._config) as db:
+                olf_fw_obj = db.get_firmware(uid)
+        fw = convert_analysis_task_to_fw_obj(analysis_task, base_fw=olf_fw_obj)
         with ConnectTo(InterComFrontEndBinding, self._config) as sc:
-            sc.add_re_analyze_task(fw)
+            sc.add_re_analyze_task(fw, unpack=re_do)
```

Comment:
Reading the code, I do not quite understand what the `olf` in `olf_fw_obj` means here. Did you mean `old`? Maybe `base_fw` would be more suitable.

Reply:
Yes, that was a typo :rofl: -> fixed